cli/src/pcluster/resources/imagebuilder/custom_component_script_https.yaml

```diff
@@ -0,0 +1,49 @@
+name: ScriptComponent
+description: ScriptComponent
+schemaVersion: 1.0
+
+phases:
+  - name: build
+    steps:
+      # Exit code for failures
+      - name: Fail
+        action: ExecuteBash
+        inputs:
+          commands:
+            - echo 1
+
+      # Script name
+      - name: ScriptName
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              echo {{script-name}}
+
+      # Script url
+      - name: ScriptUrl
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              echo {{script-url}}
+
+      # Download bash script
+      - name: Download
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              curl --retry 3 -L -o /tmp/{{ build.ScriptName.outputs.stdout }} {{ build.ScriptUrl.outputs.stdout }}
+
+      # Execute bash script
+      - name: Execute
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              (sudo chmod +x /tmp/{{ build.ScriptName.outputs.stdout }} && sudo /tmp/{{ build.ScriptName.outputs.stdout }}) || (echo 'The bash script is not executable.' && exit {{ build.Fail.outputs.stdout }})
```

Comment:
sudo is not needed, let's check it.
I would also separate the action: first make the script executable, then execute it

Reply:
done